cmd/gardenlet/app/app.go

```diff
@@ -475,6 +482,33 @@ func cleanupLegacyLoggingManagedResource(ctx context.Context, seedClient client.
 	return flow.Parallel(taskFns...)(ctx)
 }
 
+// TODO(rfranzke): Remove this code after v1.86 has been released.
+func (g *garden) cleanupOrphanedExtensionsServiceAccounts(ctx context.Context, gardenClient client.Client) error {
+	serviceAccountList := &metav1.PartialObjectMetadataList{}
```

Comment:
Gardenlet already watches structured `ServiceAccounts` in the garden cluster (in the `tokenrequestor`).
Should we use the structured cache here instead of adding another metadata-only cache for `ServiceAccounts`?